floc.bs

```diff
@@ -37,113 +27,143 @@ spec: WebIDL; urlPrefix: https://heycam.github.io/webidl/#
+  An [=interest cohort=] might reveal sensitive information. As a first mitigation, the browser should remove sensitive categories from its data collection. But this does not mean sensitive information can’t be leaked. Some people are sensitive to categories that others are not, and there is no globally accepted notion of sensitive categories.
 
   Cohorts could be evaluated for fairness by measuring and limiting their deviation from population-level demographics with respect to the prevalence of sensitive categories, to prevent their use as proxies for a sensitive category. However, this evaluation would require knowing how many individual people in each cohort were in the sensitive categories, information which could be difficult or intrusive to obtain.
 
   It should be clear that FLoC will never be able to prevent all misuse. There will be categories that are sensitive in contexts that weren't predicted. Beyond FLoC's technical means of preventing abuse, sites that use cohorts will need to ensure that people are treated fairly, just as they must with algorithmic decisions made based on any other data today.
 
   <h3 id="tracking-people-via-their-interest-cohort">Tracking people via their interest cohort</h3>
-  A cohort could be used as a user identifier. It may not have enough bits of information to individually identify someone, but in combination with other information (such as an IP address), it might. One design mitigation is to ensure cohort sizes are large enough that they are not useful for tracking. The Privacy Budget explainer points towards another relevant tool that FLoC could be constrained by.
+  An [=interest cohort=] could be used as a user identifier. It may not have enough bits of information to individually identify someone, but in combination with other information (such as an IP address), it might. One design mitigation is to ensure cohort sizes are large enough that they are not useful for tracking. In addition, if the user agent believes that too many high-entropy bits of information have already been consumed by a given {{Document}}, then the {{Document/interestCohort()}} algorithm will return a rejected promise, which can help mitigate such tracking.
+
+  The time series of [=interest cohorts=] can also be used as a user identifier. Sites could associate users' first-party identity with a series of [=interest cohorts=] observed over time, and could report these series to a single tracking service. The tracking service could then associate each series with the sites to know the browsing history of an individual.
+
+  One design mitigation is to compute the [=interest cohort=] on the fly so that sites are less likely to observe the same [=interest cohort=] for a given user. However, one could imagine that those unstable [=interest cohorts=], when casting to a lower dimension, are still going to be stable, and the above concerns still apply though to a lesser extent.
```

Comment:
I'm not comfortable with this mitigation.  My intuition is that recalculating on the fly leads to giving out substantially _more_ information — if site X observes a person in cohort A and then shortly afterwards sees that same person in cohort B, then there is a lot of opportunity to figure out what one additional web site visit is the most likely thing to have caused this change.  Limiting the update frequency, so that each change covers a bunch of new data, seems substantially better.

Reply:
This section talks about the "tracking" risk, and the subsequent section will talk about the mitigation for the risk you described.

Also just want to point out that the baseline algorithm in this specification is still/already on-the-fly, while there are complementing paragraphs pointing out that it's allowed to use a cached floc for performance concern and/or to use a per-site cached floc for mitigating the risk of recovering the browser history for flocs.